{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6b989ab8-196b-4b5c-bf06-0d210e1550bf",
   "metadata": {},
   "source": [
    "M08: Challenge Project Computational Prediction Tasks for Structure\n",
    "# Computational Strategies for Structural Predictions Task of the RYR1 Protein\n",
    "Kathryn Morrison  \n",
    "Protein Bioinformatics  \n",
    "Johns Hopkins University, 2025\n",
    "\n",
    "## Introduction\n",
    "The Ryanodine Receptor 1 (RYR1) is a critical protein within the skeletal muscle excitation-contraction coupling (ECC) complex within vertebrates. The ECC links the electrical signaling from neurons into mechanical contraction of muscle fibers. RYR1 works closely with Cav1.1 (a.k.a. DHPR), STAC3, JPH1/2 and other proteins to form a multi-protein complex within the triad junctions of muscle fibers to regulate calcium signaling. RYR1 is a large calcium channel that quickly releases calcium ions from the sarcoplasmic reticulum into the cytoplasm of muscle cells in order to trigger contractile proteins downstream (Bolaños & Calderón, 2022; Woll & Van Petegem, 2022).\n",
    "\n",
    "It is essential to study protein structure using predictive computational tools such as homology modeling, structural alignment, and secondary structural predction to better understand their function and role in biological processes. Computational strategies will be employed to model the structure of RYR1, identify conserved regions, and assess stability of secondary elements followed by subsequent analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cab2f81c-18dd-4ae5-b584-667681c82f92",
   "metadata": {},
   "source": [
    "## Questions and \"Need to Knows\"\n",
    "**1. How can template selection improve homology modeling accuracy for large proteins like RYR1?**\n",
    "\n",
    "- How do we evaluate the quality of potential template structures for homology modeling?\n",
    "- What are the main challenges when selecting a template for large, multi-domain proteins?\n",
    "- How does template selection affect homology modeling, especially in large proteins?\n",
    "- What computational tools are currently used for homology modeling?\n",
    "- Are there specific computational tools that are used for the homology modeling of large proteins?\n",
    "\n",
    "\n",
    "**2. How can structural alignment be used to identify conserved structural features within RYR1 domains?**\n",
    "\n",
    "- What algorithms and computational tools can perform structural alignment between subunits of the same protein?\n",
    "- How does structural alignment find similarity between structures?\n",
    "- How do structural alignment tools score how conserved a structural region is?\n",
    "- How do we define if a region is structurally conserved?\n",
    "- How can we visualize conserved regions from structural alignment output?\n",
    "\n",
    "\n",
    "**3. Are there differences in how alpha helices and beta sheets contribute to the structural stability of the RYR1 protein?**\n",
    "\n",
    "- What are molecular interactions that stabilize alpha helices and beta sheets?\n",
    "- What computational tools can assess how alpha helices and beta sheets are distributed and organized throughout the protein?\n",
    "- Are alpha helical or beta sheet-rich regions found more in flexible regions of the protein?\n",
    "- What computational tools can we use to quantify the stabilization provided by alpha helices and beta sheets?\n",
    "\n",
    "> Three complex questions where developed to address homology modeling challenges, structural alignment, and secondary structure stability prediction. \"Need to Knows\" that can be answered either partially or in full from the material submitted in Part 1 include those that cover RYR1's structure, including its subunits and secondary and tertiary structure, as well as template selection for such a large protein. \"Need to Knows\" involving the evaluation of computational tools such as homology modeling, structural alignment, and protein visualization that can be answered by the course material. The course material also provides knowledge as to how molecular interactions and dynamics, as well as structural elements, are known and defined. I will need outside information to address the \"Need to Knows\" involving computational tools for homology modeling of large proteins and those aligning subunits of the same protein, as well as the challenges presented when doing homology modeling on large proteins. Additionally, the \"Need to Knows\" addressing the distribution and stabilization of secondary structures in Question 3 will likely need outside information alongside course and referenced material.\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a878c9b-8043-4727-b0c3-da7acf695a3c",
   "metadata": {},
   "source": [
    "## Analysis: "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "611427b1-c064-406b-949d-a1831cc95e09",
   "metadata": {},
   "source": [
    "## *1) How can template selection improve homology modeling accuracy for large proteins like RYR1?*\n",
    "\n",
    ">Homology modeling predicts the 3D structure of a protein by comparing its sequence to a known structure, or a \"template\". The accuracy of a predicted model depends on the quality of the template used- good templates should have strong experimental quality (like resolution in Å) and alignment, high sequence identity to the target, GMQE, and structural similarity. Template quality can be evaluated based on predicted local similarity to the target and overall model quality (Waterhouse et al., 2018).  \n",
    ">\n",
    ">RYR1 contains over 5,000 amino acids. Large multi-domain proteins like RYR1 contain domains that may not all exist in one quality, single template for homology modeling. Different domain orientations, flexible regions, conformations, and template availability affect the accuracy of predicted models for large proteins (Waterhouse et al., 2018; Bolaños & Calderón, 2022).  \n",
    ">\n",
    ">The most common computational tools used for homology modelling include SWISS-MODEL, MODELLER, and Phyre. However, some tools, like SWISS-MODEL, cannot model target sequences longer than 5,000 residues. A recent tool, DEMO2, specializes in homology modeling of multi-domain proteins by applying deep-learning predicted inter-domain restraints and template alignments (Zhou et al., 2022). Oftentimes, the common models listed above are used by breaking the target into its multiple domains, and then comparing each domain to a separate template. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7b21985-2acc-4ff6-9b8a-a4e8b6304a78",
   "metadata": {},
   "source": [
    "#### **Step 1: Obtain RYR1 Sequence Information**\n",
    "\n",
    "The full FASTA sequence of [PDB entry 5GKY](https://www.rcsb.org/structure/5GKY) for the RYR1 protein was retrieved and used as the target sequence because it is proven to be the C1 conformer in a closed state with no mutations, which minimizes template noise (Bai et al., 2016).\n",
    "![Screenshot](images/Screenshot%202025-10-23%20155259.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3ef845d-2b74-41a7-b13a-b81efa8a73f6",
   "metadata": {},
   "source": [
    "#### **Step 2: Template Search and Generate Models using Phyre2**\n",
    "*SWISS-MODEL cannot model target sequences longer than 5,000 aa (like RYR1), and the DEMO2 server has been offline for an unknown number of days. MODELLER could not work on this computer.  \n",
    "Phyre2 was therefore used to generate two predictive models, as it has a capacity for modeling large sequences like RYR1.\n",
    "\n",
    "<u>Input:</u>   \n",
    "Target sequence [5GKY](https://www.rcsb.org/structure/5GKY) was input into the search query of the browser-based Phyre2 tool (Kelley et al., 2015).  \n",
    "\n",
    "<u>Output:</u>  \n",
    "[Phyre Output Files](images/d93b978a3eaf5ed7.tar.gz)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9002cd80-3719-418d-ac40-cb08667695d3",
   "metadata": {},
   "source": [
    "#### **Step 3: Choose Templates and Generate Models**\n",
    "\n",
    ">An excellent (highest identity, coverage, and confidence), a good (moderate quality), and a relatively low quality (low identity or confidence) template was selected to see how each would affect the accuracy of the predicted model it produces in comparison to the target. Confidence scoring, sequence identity, and coverage were all determining factors used to assess the quality of each template.\n",
    ">\n",
    ">A [Phyre2 output tarball file](images/d93b978a3eaf5ed7.tar.gz) was produced for all templates, including the three chosen templates. This tarball includes a modelling summary for the given sequence, as well as aligned homology models that provide confidence scores (% identity, coverage, and often predicted secondary structure). Each template has a .PDB, .fasta, .png, and .scop file.\n",
    "\n",
    "________________________________________________________________________________________\n",
    "\n",
    "EXCELLENT TEMPLATE (top model):\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20175548.png)\n",
    "\n",
    "Corresponding predicted model:  \n",
    "![Excellent Model Image](images/question1/c8vjkA_.1.big.png)\n",
    "\n",
    "________________________________________________________________________________________\n",
    "\n",
    "GOOD TEMPLATE:\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20180334.png)\n",
    "\n",
    "Corresponding predicted model:  \n",
    "![Good Model Image](images/question1/c9c1eB_.3.big.png)\n",
    "\n",
    "________________________________________________________________________________________\n",
    "\n",
    "LOW QUALITY TEMPLATE:\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20180143.png)\n",
    "\n",
    "Corresponding predicted model:  \n",
    "![Poor Model Image](images/question1/c5xa1A_.11.big.png)\n",
    "\n",
    "________________________________________________________________________________________\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd234023-4c5c-4721-b120-655815263acf",
   "metadata": {},
   "source": [
    "## *2) How can structural alignment be used to identify conserved structural features within RYR1 domains?*\n",
    "\n",
    ">Structural alignment is used to compare the 3D structures of proteins or protein domains to find structural similarities. Structural alignment tools often use backbone atom coordinates, superimposition, and distance matrices to compare two structures.  Aligning different subunits/domains of a large, multidomain protein like RYR1 can be used to determine if there are structural features or motifs that are found between two or more domains. Common tools for structural alignment include TM-align, DALI, SSM, and RCSB PBD Pairwise Structure Alignment Tool.  \n",
    ">\n",
    ">Scoring matrices for comparison include RMSD, TM-score, Z-score, and Q-score. Conserved motifs are often functional and stable regions that are likely evolutionarily conserved. A structurally conserved region often has great supimposition with a strong secondary elements maintained, usually scoring a low local RMSD, a Q-score or TM-score closest to 1, or a high Z-score (RCSB Protein Data Bank, n.d.)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49d0e382-8188-4e1f-a701-e7368d7d4cca",
   "metadata": {},
   "source": [
    "#### **Step 1: Obtain RYR1 Domain Sequence Information**\n",
    "\n",
    "The full FASTA sequence [PDB entry 5GKY](https://www.rcsb.org/structure/5GKY) for the RYR1 protein was retrieved and used, as in the previous question; it is the C1 conformer in a closed state with no mutations. Chain A was selected. (Bai et al., 2016)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b90aa7ff-b492-450c-96b5-18998c47e35f",
   "metadata": {},
   "source": [
    "#### **Step 2: Find Coordinates of RYR1 Domains**\n",
    "\n",
    "Three RYR1 domains from the same Chain A were selected from the Annotations page of the [PDB entry 5GKY](https://www.rcsb.org/structure/5GKY), under \"Domain Annotation: ECOD Classification\" (Cheng et al., 2014). Each domain identifier within the table provides coordinates to the residues for each domain. The RYR, SPRY, and MIR domains were chosen, as they are all sequentially far from each other, and likely do not have identical functions. Therefore, conserved regions will be more pronounced (as opposed to using structures with high similarity/ conservation).\n",
    "\n",
    "> <u>Domain Coordinates & Identifiers</u>  \n",
    "    **MIR**: 206-394 ([e5gkyA11](http://prodata.swmed.edu/ecod/complete/domain/e5gkyA11))   \n",
    "    **SPRY**: 1070-1246 ([e5gkyA12](http://prodata.swmed.edu/ecod/complete/domain/e5gkyA12))    \n",
    "    **RYR**: 2374-2940 ([e5gkyA8](http://prodata.swmed.edu/ecod/complete/domain/e5gkyA8))    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e01ae407-5d7d-4155-8c40-c106dffb0e0e",
   "metadata": {},
   "source": [
    "#### **Step 3: Execute Structural Alignment of RYR1 Domains**\n",
    "\n",
    "To test if conservation exists across internal domains of RYR1, the [RCSB PDB Pairwise Structural ALignment Tool](https://www.rcsb.org/alignment) was used. Three RYR1 domains were input using \"Chain A\" of the PDB entry [5GKY](https://www.rcsb.org/structure/5GKY) and the coordinates found above (Bittrich et al., 2024).\n",
    "\n",
    "<u>Input:</u>\n",
    "\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20172917.png)\n",
    "\n",
    "<u>Output:</u>\n",
    "\n",
    "[Download FASTA Alignment](images/5gky_q1.fasta)\n",
    "\n",
    "![Screenshot1](images/Screenshot%202025-10-24%20003353.png)\n",
    "\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20172818.png)\n",
    "![Screenshot1](images/Screenshot%202025-10-21%20173212.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "786e7a43-4557-413c-8e4a-9837822d3e52",
   "metadata": {},
   "source": [
    "### *3) Are there differences in how alpha helices and beta sheets contribute to the structural stability of the RYR1 protein?*\n",
    "\n",
    ">Alpha helices and beta sheets are two types of secondary structures within proteins. Alpha helices are formed by intramolecular hydrogen bonds within a single polypeptide chain, and beta strands are formed by intermolecular hydrogen bonds between two or more polypeptide strands. Both alpha helices and beta sheets have stabilizing hydrophobic interactions between side chains, as well as salt bridges that influence overall stability. Alpha helices also experience stabilizing \"capping\" at each end. Buried helices and sheets are more stable than when exposed because they are within the protein core and less accessible to solvent, Solvent Accessible Surface Area (SASA) is the total surface area of a molecule that is exposed to solvents (Kabsch & Sander, 1983; Schrödinger 2023).\n",
    ">\n",
    ">Computational tools like DSSP and PyMol, Mol*Viewer, and ECOD can assess how these secondary structures are distributed and organized throughout the protein through the identification of visualization of these structures. Classification databases like CATH, SCOPE, and PDBsum provide limited to no documentation for the PDB entry 5GKY, as RYR1 is a large protein. Other computational tools, such as STRIDE, GROMACS, and Rosetta, can identify the distribution of secondary structures, as well as locally quantify and evaluate the stability provided by these secondary structures. PyMol and other tools can provide solvent accessibility data for the protein (buried vs. exposed), giving insight into the structural stability of the secondary structure."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e88944d7-494d-4db2-8f51-5565fdb7fc80",
   "metadata": {},
   "source": [
    "#### **Step 1: Obtain RYR1 Sequence Information**\n",
    "The full FASTA sequence [PDB entry 5GKY](https://www.rcsb.org/structure/5GKY) for the RYR1 protein was retrieved and used, as in the previous questions; it is the C1 conformer in a closed state with no mutations (Bai et al., 2016)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28fcb98e-580c-41df-a1d3-da017729da77",
   "metadata": {},
   "source": [
    "#### **Step 2: Assess Secondary Structure Distribution within RYR1**\n",
    "\n",
    "PDBsum did not have secondary structure analyses for 5GKY (Laskowski 2021).\n",
    "\n",
    ">![Screenshot1](images/Screenshot%202025-10-23%20170712.png)\n",
    "\n",
    "There were no 5GKY entries in either the CATH or SCOPE database (Silitoe et al., 2021; Andreeva et al., 2020).\n",
    ">![Screenshot1](images/Screenshot%202025-10-23%20171533.png)  \n",
    ">![Screenshot1](images/Screenshot%202025-10-23%20171703.png)  \n",
    "\n",
    "<u>The ECOD database was utilized, with \"5GKY\" input into the search query of the [ECOD database website](http://prodata.swmed.edu/ecod/).</u>   \n",
    ">There were ECOD entries for all 13 domains of Chain A within the PDB entry 5GKY. This provides an overview of which type of secondary structure (alpha helices vs. beta sheets) each domain is primarily made up of, and corresponding coordinates (Cheng et al., 2014).\n",
    "  \n",
    "![Screenshot1](images/Screenshot%202025-10-23%20172626.png)  \n",
    "![Screenshot1](images/Screenshot%202025-10-23%20172645.png)\n",
    "\n",
    "<u>Visualization of Secondary Structure Distribution via Mol* Viewer</u> \n",
    "Using Mol* Viewer, I was able to assign secondary structure to RYR1.\n",
    ">- Within the PDB entry for 5GKY, there is a 3D model of the structure under the \"Structure\" tab  \n",
    ">- Under \"Components\" there is a Polymer component. By clicking on the \"...\" on the right of this line, there is a dropdown menu  \n",
    ">- Select \"Set Coloring\" within the dropdown menu  \n",
    ">- Under \"Residue Property\" select \"Secondary Structure\"  \n",
    ">- This was color alpha helices as pink, beta sheets as yellow, and loops/ disordered regions as white\n",
    "\n",
    "![Screenshot1](images/Screenshot%202025-10-23%20175903.png)\n",
    "\n",
    "<u>Using Secondary Structure Prediction Tool</u>  \n",
    "\n",
    "I was unable to assign secondary structure to RYR1 using DSSP because the protein is too large (Touw et al., 2015; Kabsch & Sander, 1983).\n",
    ">![Screenshot1](images/Screenshot%202025-10-23%20191532.png)\n",
    "\n",
    "Using STRIDE, I was unable to assign secondary structure to RYR1 because the PDB file for 5GKY is either incomplete or in an incorrect format that STRIDE cannot read \n",
    "(Heinig & Frishman, 2004).\n",
    ">![Screenshot1](images/Screenshot%202025-10-23%20190614.png)\n",
    "\n",
    "\n",
    "\n",
    "Eventually, PyMol was chosen- it successfully assigned secondary structure, while using Solvent-Accessible Surface Area (SASA) to determine local stability (Schrödinger 2023).  \n",
    "<u>PyMol Code:</u>  \n",
    ">load 5gky.cif  \n",
    "dss  \n",
    "color pink, ss h  \n",
    "color yellow, ss s  \n",
    ">\n",
    ">select helix, ss h  \n",
    "select beta, ss s  \n",
    ">\n",
    ">get_area helix   \n",
    "get_area beta  \n",
    " \n",
    "Below is a code snippet of the output\n",
    "![Screenshot1](images/Screenshot%202025-10-23%20212225.png)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1398d2ff-7b25-4e4c-95e4-3545343ac24d",
   "metadata": {},
   "source": [
    "#### **Step 3: Assess Stability Contributions of Secondary Structures**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c955899-de67-4a1f-b14a-4ec4f6b6a771",
   "metadata": {},
   "source": [
    "To calculate SASA per residue for each secondary structure:\n",
    ">\n",
    ">\n",
    ">select helices, ss h  \n",
    "select sheets, ss s  \n",
    ">\n",
    ">helix_resi = set()d  \n",
    "iterate helices, helix_resi.add(resi)  \n",
    "num_helix = len(helix_resi)  \n",
    ">\n",
    ">beta_resi = set()d  \n",
    "iterate sheets, beta_resi.add(resi)  \n",
    "num_beta = len(beta_resi)  \n",
    ">\n",
    ">av_helix = (get_area helix)/ num_helix  \n",
    ">av_beta = (get_area beta)/ num_beta  \n",
    ">\n",
    ">print (av_helix)  \n",
    ">print(av_beta)  \n",
    " \n",
    "Below is the output\n",
    "![Screenshot1](images/Screenshot%202025-10-23%20223602.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "118bfb45-85d4-42e0-ae5f-e55c90608769",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "#### **Question 1:**\n",
    "\n",
    "The Phyre2 computational tool produced predicted models for RYR1 based on template models of varying quality. Three were chosen, and all had 100% confidence. Phyre2 (with normal settings) only generates models for the tope templates that pass a certain confidence (>90%) and coverage threshold. The excellent (top) model had the highest coverage (85%), showing that this is a reliable match to the template.   \n",
    "\n",
    "\n",
    "<u>Table of Chosen Template Quality Results</u>\n",
    "| Chosen Template Quality | Template ID | Confidence Score | Sequence Identity (%) | Coverage (%) |\n",
    "|-------------------------|-------------|------------------|-----------------------|--------------|\n",
    "| Excellent               |c8vjkA       |100               |98                     |87            |\n",
    "| Good                    |c9c1eB       |100               |72                     |80            |     \n",
    "| Low Quality             |c5xa1A       |100               |20                     |23            |\n",
    "\n",
    "\n",
    "#### **Question 2:**\n",
    "\n",
    "RCSB PDB Pairwise Structural Alignment Tool was used to structurally align three different domains of Chain A in RYR1. This alignment table revealed a structural motif within residues 295-310aa of the sequence alignment. Visual inspection using the Mol* Viewer within the structural alignment tool indicated that the partially conserved region through all three domains was an alpha helix backbone and a few of its stabilizing side chain residues. \n",
    "\n",
    "![Screenshot1](images/Screenshot%202025-10-24%20011122.png)  \n",
    "Position: 306 aa (Bittrich et al., 2024). \n",
    "\n",
    "\n",
    "#### **Question 3:**\n",
    "\n",
    "Using the classification database ECOD identified alpha-helical and beta-sheet rich regions within all 13 domains of Chain A in RYR1. Alpha-helical regions were found in the many repetitive alpha hairpins (e5gkyA2, e5gkyA3, e5gkyA4, e5gkyA6) and voltage-gated, transmembrane ion channels (e5gkyA1).  Beta-sheet-rich domains were found in the beta-Trefoil (e5gkyA10, e5gkyA11) and jelly-roll domains (e5gkyA12, e5gkyA5, e5gkyA7). \n",
    "\n",
    "Secondary structure distribution of RYR1 tetramer was mapped and identified using MolViewer and PyMol. These computational tools identified and visualized the locations of alpha helices and beta sheets within the protein. To assess stability contributions of the secondary structures, the solvent-accessible surface area (SASA) was calculated per residue for both alpha helices and beta sheets. The total SASA value for alpha helices was 815,795 Å²,and 141,002 Å² for beta sheets. The number of residues for each secondary structure was totaled to find the average SASA per residue:  \n",
    "Alpha helices = 359.54 Å²/residue  \n",
    "Beta sheets = 377.01 Å²/residue"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39a9c1da-1bab-4c54-947a-7738573c8e10",
   "metadata": {},
   "source": [
    "## Discussion\n",
    " \n",
    "The results for the first question demonstrate that template selection strongly influences homology-modeling accuracy. Specifically, selecting a high-quality template is shown to improve homology modeling. Comparison of these three templates using Phyre2 demonstrates an approach for determining the sensitivity and accuracy of homology models that can be extended to other large, multi-domain proteins.\n",
    "\n",
    "The excellent (top) model produced by Phyre2 used the highest quality template c8vjkA, maintaining the secondary and tertiary structure of the original target (5GKY) and producing a better predicted model, characterized by higher sequence identity and coverage between the two models. This, in turn, better preserves inter- and intra-domain organization in large, multi-domain proteins like RYR1. The low-quality template yielded slightly less favorable results, producing a predicted model with low sequence identity and low coverage. The homology modeling accuracy is very unfavorable below a 30% sequence identity, as shown in the low-quality model c5xa1A. Additionally, these results align with previous literature that sequence identity and coverage from a template must be above 40% to maintain most of the tertiary topology within a predicted structure (Waterhouse et al., 2018). Further studies should be done in addition to this experiment, where the number of templates being compared is increased and different homology modeling tools are utilized with the same target sequence. Homology modeling tools vary in how they model their templates (mathematical approach, file output, contents of database used, etc), which should be accounted for and compared. \n",
    "\n",
    "The MSA output table from the RCSB PDB Pairwise Structural Alignment Tool from question 2 uses purple shading, with darker purple indicating higher similarity and close superimposition, and light purple indicating partial similarity. The 3D viewer can then be used to assess the conserved, dark purple regions. There was very little conservation between all three structures (only a short, partial local structural feature), likely because there was only a span of ~200aa that could be structurally aligned due to the short length of the MIR domain. Additionally, the domains of RYR1 are likely to have different structural components, as different domains have evolved unique folds/ tertiary structures that can carry out specific functions. Therefore, it is not surprising that there is little conservation between the three domains. The finding of limited conservation between inter-domains of a protein is consistent with Bittrich et al. (2024), a study that stated RYR1's structural diversity across domains allows for different functional mechanisms to occur. Further alignment using TM-align or DALI should be used to quantify conservation more locally and in greater detail. \n",
    "\n",
    "Analyses within this document suggest that alpha helices and beta sheets contribute differently to the stability of the RYR1 protein. Many different computational tools were tested to identify the distribution of secondary structures within the RYR1 protein, but the large PDB file size of the protein (also containing the 4 chains that make up this large ~20,000aa tetrameric protein) was proving to be difficult for browser-based applications like DSSP and STRIDE. Such processes require a lot of memory, and they are often not designed to handle multiple chains at a time. These two tools would have been better and more able to directly quantify the stability of secondary structures in the RYR1 protein than PyMol. Other databases struggled with this large protein as well (PDBsum, CATH, and SCOPE). \n",
    "\n",
    "The domains from the ECOD database were given architectural assignment. For each chain, there are many repetitive alpha hairpin domains that may serve as scaffolding for protein-protein interactions amongst the chains of the tetrameric assembly. One alpha-helical region makes up the transmembrane ion channel domain, where alpha helices are likely to span the membrane and form a hydrophobic pore for calcium to flow. The globular beta-Trefoils within the two domains of RYR1 have a compact arrangement of interlocking beta-strands that form a barrel/core that is highly stable and resistant to folding, while maintaining overall domain shape. Beta sheet-rich jelly roll domains within RYR1 also maintain the shape of the protein, acting as a backbone. These domains are commonly repeated in large protein assemblies to maintain structural integrity (Fox et al., 2014; Chandonia et al., 2022).\n",
    "\n",
    "PyMol was chosen, as it is a more scalable approach that can still help approximate the stability of alpha helices and beta sheets within RYR1 using SASA- an indirect measure of local stability. Low SASA is indicative of secondary structures being more buried, and therefore more stable, whereas higher SASA indicates that secondary structures are more exposed and therefore less stable. The analysis performed for Question 3 indicates that the alpha helices in RYR1 have a lower average SASA (~ 359 Å²) per residue than their beta sheet counterparts (~ 377 Å²), suggesting that they are more buried in the core and therefore more stable. Alpha helices are often part of the stabilization of the hydrophobic core of proteins and are resistant to folding. The beta sheets in the RYR1 protein are slightly more exposed, suggesting that they provide structural rigidity (ex. shape) instead of being part of core stabilization. These findings align with how often beta sheets are seen on protein surfaces, where they can provide rigidity and support. These findings are consistent with Fox et al. (2014), which notes that greater burial of alpha helices supports hydrophobic core stability of large calcium channel proteins. \n",
    "\n",
    "It is important to note that SASA is an indirect measure of protein stability. Further analysis for this question could be done with more detailed computational approaches (like Rosetta) that can better assess local residue stability and directly quantify energy contributions of secondary structures within the protein. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea6c48c4-9d8e-4337-9ebf-3a8190a1da45",
   "metadata": {},
   "source": [
    "## References\n",
    "\n",
    "Andreeva, A., Kulesha, E., Gough, J., & Murzin, A. G. (2020). The SCOP database in 2020: Expanded classification of representative family and superfamily domains of known protein structures. Nucleic Acids Research, 48(D1), D376–D382. https://doi.org/10.1093/nar/gkz1064\n",
    "\n",
    "Bai, X.-C., Yan, Z., Wu, J., Li, Z., & Yan, N. (2016). The central domain of RyR1 is the transducer for long‑range allosteric gating of channel opening. Cell Research, 26(8), 995–1006. https://doi.org/10.1038/cr.2016.89\n",
    "\n",
    "Berman, H. M., et al. (2000). The Protein Data Bank. Nucleic Acids Research, 28(1), 235–242.\n",
    "\n",
    "Bittrich, S., Segura, J., Duarte, J. M., Burley, S. K., & Rose, Y. (2024). RCSB Protein Data Bank: Exploring protein 3D similarities via comprehensive structural alignments. Bioinformatics, 40(6), btae370. https://doi.org/10.1093/bioinformatics/btae370\n",
    "\n",
    "Bolaños, P., & Calderón, J. C. (2022). Excitation-contraction coupling in mammalian skeletal muscle: Blending old and last-decade research. Frontiers in Physiology, 13, 989796. https://doi.org/10.3389/fphys.2022.989796\n",
    "\n",
    "Chandonia, J. M., Guan, L., Lin, S., Yu, C., Fox, N. K., & Brenner, S. E. (2022). SCOPe: Improvements to the Structural Classification of Proteins—extended Database to facilitate Variant Interpretation and Machine Learning. Nucleic Acids Research, 50(D1), D553–D559. https://doi.org/10.1093/nar/gkab1054\n",
    "\n",
    "Cheng, H., Schaeffer, R. D., Liao, Y., Kinch, L. N., Pei, J., Shi, S., et al. (2014). ECOD: An evolutionary classification of protein domains. PLoS Computational Biology, 10(12), e1003926. https://doi.org/10.1371/journal.pcbi.1003926\n",
    "\n",
    "Fox, N. K., Brenner, S. E., & Chandonia, J. M. (2014). SCOPe: Structural Classification of Proteins—extended, integrating SCOP and ASTRAL data and classification of new structures. Nucleic Acids Research, 42(D1), D304–D309. https://doi.org/10.1093/nar/gkt1240\n",
    "\n",
    "Heinig, M., Frishman, D. (2004). STRIDE: a Web server for secondary structure assignment from known atomic coordinates of proteins. Nucl. Acids Res. , 32, W500-2.\n",
    "\n",
    "Kabsch, W., & Sander, C. (1983). Dictionary of protein secondary structure: Pattern recognition of hydrogen-bonded and geometrical features. Biopolymers, 22(12), 2577–2637. https://doi.org/10.1002/bip.360221211\n",
    "\n",
    "Kelley, L. A., Mezulis, S., Yates, C. M., Wass, M. N., & Sternberg, M. J. E. (2015). The Phyre2 web portal for protein modeling, prediction and analysis. Nature Protocols, 10(6), 845‑858. https://doi.org/10.1038/nprot.2015.053\n",
    "\n",
    "Laskowski, R. A. (2001). PDBsum: Summaries and analyses of PDB structures. Nucleic Acids Research, 29(1), 221–222\n",
    "\n",
    "RCSB Protein Data Bank. (2019, December 5). Introducing Mol*. https://www.rcsb.org/news/5dea9030a5007a04a313ed90\n",
    "\n",
    "RCSB Protein Data Bank. (n.d.). Pairwise structure alignment: Help documentation. RCSB PDB. Retrieved September 26, 2025, from https://www.rcsb.org/docs/tools/pairwise-structure-alignment\n",
    "\n",
    "Schrödinger, LLC. (2023). The PyMOL Molecular Graphics System (Version 3.0) [Computer software]. https://pymol.org\n",
    "\n",
    "Sillitoe I, Bordin N, Dawson N, Waman VP, Ashford P, Scholes HM, Pang CSM, Woodridge L, Rauer C, Sen N, Abbasian M, Le Cornu S, Lam SD, Berka K, Varekova IH, Svobodova R, Lees J, Orengo CA. CATH: increased structural coverage of functional space. Nucleic Acids Res. 2021 Jan 8;49(D1):D266-D273. https://doi.org/10.1093/nar/gkaa1079.\n",
    "\n",
    "Touw, W. G., Baakman, C., Black, J., te Beek, T. A. H., Krieger, E., Joosten, R. P., & Vriend, G. (2015). A series of PDB-related databases for everyday needs. Nucleic Acids Research, 43(D1), D364–D368. https://doi.org/10.1093/nar/gku1068\n",
    "\n",
    "Woll, K.A., & Van Petegem, F. (2022). Calcium-release channels: structure and function of IP3 receptors and ryanodine receptors. Physiol Rev., 102(1):209-268. https://doi.org/10.1152/physrev.00033.2020\n",
    "\n",
    "Zhou, X., Peng, C., Zheng, W., Li, Y., Zhang, G., & Zhang, Y. (2022). DEMO2: Assemble multi-domain protein structures by coupling analogous template alignments with deep-learning inter-domain restraint prediction. Nucleic acids research, 50(W1), W235–W245. https://doi.org/10.1093/nar/gkac340\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9f2e7cc6-92b6-45b4-b1ab-ef11efae849d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/katiem'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1c4606f-1b6b-478b-812b-37e6ea331851",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
